### PR TITLE
Add autoprefixer and cssnano support

### LINKED ISSUE #143: Add autoprefixer support?
I guess depends on what level of browser support we want in #101...

But if the goal is to support the _n_ most recent versions of popular browsers, we may want to use something like [**autoprefixer**](https://github.com/postcss/autoprefixer#cli) to make sure our CSS is vendor prefixed.

Doesn't look like we'd need to add Gulp or Grunt or any expensive toolchain, and can get away with a simple 2 dependencies ([**postcss-cli**](https://github.com/postcss/postcss-cli#options) and **autoprefixer**) and a npm script in package.json:

```sh
> npm install postcss-cli autoprefixer -D
> $(npm bin)/postcss public/*.css --use autoprefixer -r --no-map
```

---

<details>
<summary>Current diff:</summary>

```diff
diff --git a/public/main.css b/public/main.css
index d97ea06..0c6dd7e 100644
--- a/public/main.css
+++ b/public/main.css
@@ -6,11 +6,21 @@ html {
   font-style: normal;
   background-size: contain;
   height: 100%;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
-  align-content: center;
-  flex-direction: column;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
+  -ms-flex-line-pack: center;
+      align-content: center;
+  -webkit-box-orient: vertical;
+  -webkit-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
 }
 
 input, select, textarea, button {
@@ -39,10 +49,19 @@ input, select, textarea, button {
   width: 470px;
   height: 250px;
   border-radius: 5px;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
-  flex-direction: column;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
+  -webkit-box-orient: vertical;
+  -webkit-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
   text-align: center;
 }
 
@@ -130,6 +149,7 @@ td {
   bottom: 125%;
   left: 50%;
   margin-left: -80px;
+  -webkit-transition: opacity 0.5s;
   transition: opacity 0.5s;
   opacity: 0;
 }
@@ -155,6 +175,13 @@ td {
 #progress-bar {
   width: 300px;
   height: 5px;
+  background: -webkit-gradient(
+    linear,
+    left top, right top,
+    from(#FD9800),
+    color-stop(#D73000),
+    to(white)
+  );
   background: linear-gradient(
     90deg,
     #FD9800,
@@ -170,18 +197,33 @@ td {
   margin: 0 auto;
   width: 470px;
   height: 250px;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
 }
 
 #share-window-r {
   width: 50%;
   margin: 0 auto;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
-  flex-direction: column;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
+  -webkit-box-orient: vertical;
+  -webkit-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
 }
 
 #share-window-r > div {
@@ -190,8 +232,11 @@ td {
 }
 
 #copy {
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  flex-wrap: nowrap;
+  -ms-flex-wrap: nowrap;
+      flex-wrap: nowrap;
 }
 
 #link {
@@ -247,10 +292,19 @@ td {
   margin: 0 auto;
   width: 470px;
   height: 250px;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
-  flex-direction: column;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
+  -webkit-box-orient: vertical;
+  -webkit-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
   text-align: center;
 }
 
@@ -262,10 +316,19 @@ td {
   margin: 0 auto;
   width: 470px;
   height: 250px;
+  display: -webkit-box;
+  display: -ms-flexbox;
   display: flex;
-  justify-content: center;
-  align-items: center;
-  flex-direction: column;
+  -webkit-box-pack: center;
+      -ms-flex-pack: center;
+          justify-content: center;
+  -webkit-box-align: center;
+      -ms-flex-align: center;
+          align-items: center;
+  -webkit-box-orient: vertical;
+  -webkit-box-direction: normal;
+      -ms-flex-direction: column;
+          flex-direction: column;
   text-align: center;
 }
```
</details>

---

**NOTE:** Due to **postcss**/**autoprefixer**'s baffling indentation choices, the autoprefixed versions of the CSS will fail the CSS linter, so we'd need to run the **prettier** formatter against the CSS file after autoprefixing it (which can currently be done via `npm run format`).

### PR DESCRIPTION
**Pull request by pdehaan**
Adds autoprefixer support and cssnano minification.

Fixes #143

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ public/polyfill.min.js
 static/*
 !static/info.txt
 test/frontend/bundle.js
+public/main.css
```
Comment (pdehaan) [inline .gitignore]: I moved the public/main.css file to frontend/src/main.css. Then `npm run build:css` does the whole autoprefixer, media query combining, minification and source map magic and copies the _processed_ file back into the old public/main.css location.

File: browserslist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+last 2 versions
+Firefox ESR
+not last 2 BlackBerry versions
+not last 2 ExplorerMobile versions
+not last 2 Opera versions
+not last 2 OperaMini versions
+not last 2 OperaMobile versions
+not last 2 UCAndroid versions
```
Comment (pdehaan) [inline browserslist]: I totally made all this up. LOLOLOL.

File: postcss.config.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const mqpacker = require('css-mqpacker');
+
+const conf = require('./server/config');
+
+const options = {
+  plugins: [autoprefixer, mqpacker, cssnano]
+};
+
+if (conf.env === 'development') {
+  options.map = { inline: true };
```
Comment (pdehaan) [inline postcss.config.js]: I only enabled sourcemaps in dev mode. We can adjust this as much as anybody cares... like maybe we dont want minification on the dev server, etc.

**Comment by dannycoates** [comment]
@pdehaan can you rebase? LGTM, but what do I know about css :man_shrugging:

**Comment by pdehaan** [comment]
Hey @bwinton, mind taking a 👁 as the highest ranking CSS Czar?

**Comment by pdehaan** [comment]
Alternatively, we could convert the source to Sass and run scss-lint against the frontend/src/*.scss, and run the stylelint compatibility check only against the compiled autoprefixer vendored output in public/*css.

**Pull request merged**